frontend/src/api/k8s/secrets.ts

```diff
@@ -26,7 +27,7 @@ export const assembleSecret = (
   const annotations: Record<string, string> = {};
 
   let stringData = data;
```

Comment:
in assembleSecret and assembleConfigmap, there is the metadata label constant named DASHBOARD_RESOURCE being set to true. But I think the label `opendatahub.io/managed: true` is set, too, I just do not know by which component (notebook controller possibly, maybe the operator itself ...) Do you see label `opendatahub.io/managed: true` in created env-related configmaps and secrets?

The types SecretKind and ConfigmapKind have a provision for labels, in this case, `opendatahub.io/dashboard: true` is set here in code i.e. that konwn labels contst DASHBOARD_RESOURCE.

The other const representing  `opendatahub.io/managed` has nothing do do with AWS related secrets, if you look around, you see all kinds of deployment objects in the main odh namespace and all sorts of other objects having that label.

I would argue against using a secret-prefix, as we cannot control users outside odh also using configmap-xxx or secret--- format.

Reply:
Yep this was the original solution. We went with this prefix solution to limit the number of api calls in the dashboard bc in the current context we were not already fetching the secret/config map. 

@andrewballantyne can chime in here becuase he was the one opting for this prefix solution over the label solution

Reply:
also to your side note...
> Small side note: currently, in the edit workbench section, shared configmaps and secrets also appear if they do not have the opendatahub.io/dashboard label. I think currently, all logic is exclusively by name and entry in the envFrom list, never by anything else like labels.

Yes that is true, we just try to fetch whatever is in envfrom, we do put that label on those secrets/configmaps on edit. do we use it, dont think so. probably just a artifact from an earlier version of the UI

Reply:
We are addressing the loss of configmaps/secrets not created by the Dashboard first by the same way we do it for DSPA secret.

This is not the final solution for addressing this, but we need to understand and plan our action to how we approach it. Dashboard is no longer in a place to address a system of resources and how one crafts or deletes them on the cluster like we were back a year or two ago. We'll need architectural design documents and walk through a general plan for all resources, not adhoc for how Workbenches handles it.

I'd like us to align on the current implementation we use for DSPA to avoid deleting secrets not generated. It's fair to say "configmap-" and "secret-" prefixes followed by a number is not ideal... but it's the easiest way to support backwards compatibility with what we have as we determine the way to move forward.

Given that in the system of injecting extra configmaps and secrets not through the UI is already on a custom adventure, it can just name them something different; and I'd argue people probably don't organically call their configmaps "configmap-<fixed length generated characters>" (I think we use 6 alphanumerical chars)

To address "but why not just make it more robust" -- we intend to. Just not with this initial effort. See my comment on the ticket, we will come back and address this more robustly in the future. But I'd prefer to align all of Dashboard on the same implementation then "we do it this way here, and that way there, and a different way when this applies". 

Hope that helps with what and why and where we plan to go after this.

Reply:
> to say "configmap-" and "secret-" prefixes followed by a number is not ideal... but it's the easiest way to support backwards compatibility with what we have as we determine the way to move forward.

Given that in the system of injecting extra configmaps and secrets not through the UI is already on a custom adventure, it can just name them something different; and I'd argue people probably don't organically call their configmaps "configmap-"

Yes, I can live with that for now :-) Thank you all for your feedback and effort. The way this is handled for now, I can just make sure my custom configmaps and secrets do not start with configmap- or secret- and I'll be ok.